### PR TITLE
Rafadana/event history

### PR DESCRIPTION
**Pull request by RafadanaM**
### Background: 
- Add event history page, refactor event content page

### Additional Packages/Dependencies:
- none

### Changes:
- lib/common/components/buttons/custom_button.dart
- lib/common/components/forms/custom_form_input_class.dart
- lib/common/components/forms/custom_search_bar.dart
- lib/common/components/forms/custom_text_field.dart
- lib/common/config/enum.dart
- lib/common/config/theme.dart
- lib/event/bloc/event_matching/event_matching_home_cubit.dart
- lib/event/bloc/event_matching/event_matching_home_state.dart
- lib/event/bloc/events_history/events_history_cubit.dart
- lib/event/bloc/events_history/events_history_state.dart
- lib/event/bloc/popular_event/popular_events_cubit.dart
- lib/event/bloc/popular_event/popular_events_state.dart
- lib/event/bloc/search_event/search_event_cubit.dart
- lib/event/bloc/search_event/search_event_state.dart
- lib/event/components/common/event_attendees.dart
- lib/event/components/common/event_card_large.dart
- lib/event/components/common/event_content_card.dart
- lib/event/components/common/event_joined_card.dart
- lib/event/data/common/event_joined_model.dart
- lib/event/data/common/event_joined_model.g.dart
- lib/event/data/common/event_joined_request_model.dart
- lib/event/data/common/event_joined_request_model.g.dart
- lib/event/data/common/event_location_model.dart
- lib/event/data/common/event_location_model.g.dart
- lib/event/data/common/event_model.dart
- lib/event/data/common/event_model.g.dart
- lib/event/data/common/event_status_model.dart
- lib/event/data/common/event_status_model.g.dart
- lib/event/data/common/filter/event_filter_model.dart
- lib/event/data/common/filter/event_filter_model.g.dart
- lib/event/data/common/filter/event_status_request_model.dart
- lib/event/data/common/filter/event_status_request_model.g.dart
- lib/event/data/common/popular_event_location_model.dart
- lib/event/data/common/popular_event_location_model.g.dart
- lib/event/data/event_detail/event_detail_model.dart
- lib/event/data/event_matching/event_matching_home_repository.dart
- lib/event/data/event_matching/event_matching_response_model.dart
- lib/event/data/event_matching/event_matching_response_model.g.dart
- lib/event/data/events_history/events_history_data_provider.dart
- lib/event/data/events_history/events_history_repository.dart
- lib/event/data/popular_event/popular_events_repository.dart
- lib/event/data/search_event/search_event_repository.dart
- lib/page/dashboard.dart
- lib/page/event/event_history.dart
- lib/page/search/search_events.dart
- lib/page/walkthrough/dummy_event_matching.dart
- lib/page/walkthrough/dummy_homepage.dart

### How to Implement:
- Use `EventJoinedCard` to create card for event history and scheduled event.  
-  Use `EventJoinedType.SCHEDULED` for scheduled event (require `onCancelHandler`)
- Use `EventJoinedType.HISTORY` for event history

### CODE REVIEW

File: lib/common/config/enum.dart [hunk 1]
```diff
@@ -292,3 +294,40 @@ enum ImageInputAction {
...
+  COMING_SOON,
+  @JsonValue("Ongoing")
+  ONGOING,
+  @JsonValue("In Progress")
+  IN_PROGRESS,
+  @JsonValue("Done")
+  DONE,
+}
+
+// extension EventStatusExtension on EventStatus {
```
Comment (YafiAhsanH) [inline lib/common/config/enum.dart]: there are still comments here

**Review by YafiAhsanH — CHANGES_REQUESTED** [review]

**Review by oliver-mf — DISMISSED** [review]
looks gud to me

**Review by YafiAhsanH — APPROVED** [review]
Looks good

**Review by oliver-mf — APPROVED** [review]
gud

**Pull request merged**